packages/auth/lib/AccessToken.js

```diff
@@ -0,0 +1,118 @@
+const crypto = require('crypto')
+const base64u = require('@orbiting/backend-modules-base64u')
+const moment = require('moment')
+const transformUser = require('./transformUser')
+
+const { newAuthError } = require('./AuthError')
+const MissingScopeError = newAuthError('missing-scope', 'api/auth/accessToken/scope/404')
+const MissingKeyError = newAuthError('missing-key', 'api/auth/accessToken/key/404')
+const MissingPackageGrant = newAuthError('missing-package-grant', 'api/auth/accessToken/pledgePackages/notAllowed')
+
+const DATE_FORMAT = 'YYYY-MM-DD'
+
+const scopeConfigs = {
+  'CUSTOM_PLEDGE': {
+    exposeFields: ['email', 'hasAddress', 'hasChargableSource', 'customPackages'],
+    pledgePackages: ['PROLONG'],
+    ttlDays: 90
+  }
+}
+
+const getScopeConfig = (scope) => {
+  const scopeConfig = scopeConfigs[scope]
+  if (!scopeConfig) {
+    throw new MissingScopeError(null, { scope })
+  }
+  return scopeConfig
+}
+
+const getHmac = (payload, key) => {
+  if (!key) {
+    throw new MissingKeyError()
+  }
+  return crypto
+    .createHmac('sha256', key)
+    .update(payload)
+    .digest('hex')
+}
+
+const getPayload = ({ userId, scope, expiresAt }) => {
+  const expiresAtFormatted = moment(expiresAt).format(DATE_FORMAT)
+  return `${userId}/${scope}/${expiresAtFormatted}`
```

Comment:
This payload would look the same with each request within a day `${userId}/${scope}/${expiresAtFormatted}`. To increase viability, would add a random component that changes with each call.

Reply:
Sorry, ignore previous comment. It's a signature, not an encryption.

Reply:
Why? That's exactly the intended use case. What would be the advantage of it looking different? I think the security would even be worse, because it would be easier to sample thousands of tokens with nearly identical content.